src/interface/save_config_file.py

```diff
@@ -12,35 +13,61 @@ def __init__(self, config_file):
         self.root = self.xmld.getroot()           
 
     def save_output_file_settings(self, new_file_path):
-        """ This function saves the changes done in UI related to output file settings.
-
-        list new_file_path -- it gets the values(path and file name ) from UI."""     
-        for output in self.root.findall('./Output'):            
-            output.find('Path').text = new_file_path[0]
-            output.find('Name').text = new_file_path[1]
-                    
-        self.xmld.write(self.config_file)   
-        
+        """Save new_file_path in configuration file.
+
+        list new_file_path --> 'd:\custompath\sudoku.txt', it gets the file path from UI."""     
+        ext_file = os.path.splitext(new_file_path)[1]
+        if ext_file =='.txt' or ext_file == '.html':
+            for output in self.root.findall('./Output'):            
+                output.find('Path').text = new_file_path            
+            self.xmld.write(self.config_file)   
+        else:
+            return False
+
     def save_algorithm_settings(self, alg_name):
-        """ This function saves the changes done in UI related to algorithm choosen to solve the game."""       
-        for alg in self.root.findall("./Solve/Algorithm"):
-            if alg.get('name') == alg_name:
-                alg.find('Status').text = 'Active'
-            else:
-                alg.find('Status').text = 'Inactive'            
+        """Save alg_name in configuration file
+
+        string alg_name --> 'Backtracking',, it gets the new algorithm name from UI."""       
+        if alg_name != '':
+            for alg in self.root.findall("./Solve/Algorithm"):
+                if alg.get('name') == alg_name:
+                    alg.find('Status').text = 'Active'
```

Comment:
always try to use variables or constants for this kind of assignations if this change from Active to True, you just need to change the variable and the result should work without problems